### PR TITLE
Fix #787 JSon XStream dataformat support

### LINKED ISSUE #787: JSon XStream dataformat support
We have a support extension (used by Salesforce, IIRC), so some parts might be there already

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #787

### CODE REVIEW

File: extensions/support/xstream/deployment/src/main/java/org/apache/camel/quarkus/support/xstream/deployment/XStreamSupportProcessor.java [hunk 1]
```diff
@@ -116,57 +89,71 @@ void process(CombinedIndexBuildItem indexBuildItem,
...
+                "java.lang.Double",
+                "java.lang.Float",
+                "java.lang.Integer",
+                "java.lang.Long",
                 "java.lang.Number",
-                "java.math.BigInteger",
-                "java.math.BigDecimal",
-                "java.lang.StringBuffer",
+                "java.lang.Object",
+                "java.lang.Object",
```
Comment (jamesnetherton) [inline extensions/support/xstream/deployment/src/main/java/org/apache/camel/quarkus/support/xstream/deployment/XStreamSupportProcessor.java]: Duplicate `java.lang.Object`.

File: extensions/support/xstream/deployment/src/main/java/org/apache/camel/quarkus/support/xstream/deployment/XStreamSupportProcessor.java [hunk 2]
```diff
@@ -116,57 +89,71 @@ void process(CombinedIndexBuildItem indexBuildItem,
...
+                "java.lang.Short",
+                "java.lang.StackTraceElement",
                 "java.lang.String",
-                "java.lang.reflect.Method",
+                "java.lang.StringBuffer",
+                "java.lang.StringBuilder",
+                "java.lang.Throwable",
+                "java.lang.Void",
+                "java.lang.Void",
```
Comment (jamesnetherton) [inline extensions/support/xstream/deployment/src/main/java/org/apache/camel/quarkus/support/xstream/deployment/XStreamSupportProcessor.java]: Duplicate `java.lang.Void`.

File: extensions/support/xstream/deployment/src/main/java/org/apache/camel/quarkus/support/xstream/deployment/XStreamSupportProcessor.java (hunk 1 above)
Comment (ppalaga) [reply extensions/support/xstream/deployment/src/main/java/org/apache/camel/quarkus/support/xstream/deployment/XStreamSupportProcessor.java]: Haha, I knew why I was sorting them!

**Comment by ppalaga** [comment]
d60af9a removed the dup reflection registrations in XStreamSupportProcessor. Thx for catching those @jamesnetherton!

File: extensions/xstream/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+---
+name: "Camel Quarkus XStream"
+description: "Camel WebSocket XStream support"
```
Comment (jamesnetherton) [inline extensions/xstream/runtime/src/main/resources/META-INF/quarkus-extension.yaml]: I guess this should be `Camel XStream support`.

**Comment by jamesnetherton** [comment]
In addition to the review comments, is the itest module referenced in the PR workflow? I don't see it.

**Comment by ppalaga** [comment]
> is the itest module referenced in the PR workflow

Good catch, it is not. And the groovy check was not executed because it was inside `<pluginManagement>` block. Adding a fix.

File: extensions/xstream/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 3 above)
Comment (ppalaga) [reply extensions/xstream/runtime/src/main/resources/META-INF/quarkus-extension.yaml]: +1

**Comment by ppalaga** [comment]
a53b54e fixes issues found by @jamesnetherton

**Pull request merged**